File: api/src/endpoint/question/vote_comment.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::VoteResource};

pub struct VoteComment;

impl Endpoint for VoteComment {
    const PATH: &'static str =
        "/repository/{owner}/{repo}/question/{number}/comment/{comment_id}/vote";
    const METHOD: http::Method = http::Method::POST;

    type Request = VoteCommentRequest;
    type Response = VoteCommentResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct VoteCommentRequest {
    pub value: i16,
}

pub type VoteCommentResponse = VoteResource;

File: gitdot-api-ts/src/endpoint/question/vote-comment.ts
import { z } from "zod";

import { VoteResource } from "../../resource";

export const VoteCommentRequest = z.object({
  value: z.number().int(),
});
export type VoteCommentRequest = z.infer<typeof VoteCommentRequest>;

export const VoteComment = {
  path: "/repository/{owner}/{repo}/question/{number}/comment/{comment_id}/vote",
  method: "POST",
  request: VoteCommentRequest,
  response: VoteResource,
} as const;
export type VoteComment = typeof VoteComment;
